src/pybind/mgr/volumes/fs/operations/resolver.py

```diff
@@ -21,9 +21,20 @@ def resolve(vol_spec, path):
 
 def resolve_trash(vol_spec, path):
     parts = splitall(path)
-    if len(parts) != 6 or os.path.join(parts[0], parts[1]) != vol_spec.subvolume_prefix or \
-       parts[4] != '.trash':
+
+    if len(parts) == 6:
+        if os.path.join(parts[0], parts[1]) != vol_spec.subvolume_prefix or \
+           parts[4] != '.trash':
+            return None
+        groupname = None if parts[2] == Group.NO_GROUP_NAME else parts[2]
+        subvolname = parts[3]
+        return (groupname, subvolname)
+    elif len(parts) == 5:
```

Comment:
Maybe add some comment like the following:
This is to resolve the trashpath of subvolumes which belong to subvolume groups with quota.  The subvolume trash path has the following syntax 
/\<vol_spec.subvolume_prefix\>/\<groupname\>/.trash/\<subvolname\> ?

Reply:
Done